OrbitVulkanLayer/Main.cpp

```diff
@@ -0,0 +1,315 @@
+}
+
+VKAPI_ATTR VkResult VKAPI_CALL OrbitEnumeratePhysicalDevices(VkInstance instance,
+                                                             uint32_t* physical_device_count,
+                                                             VkPhysicalDevice* physical_devices) {
+  VkResult result =
+      logic_.CallEnumeratePhysicalDevices(instance, physical_device_count, physical_devices);
+  logic_.PostCallEnumeratePhysicalDevices(instance, physical_device_count, physical_devices);
+  return result;
+}
+
+// ----------------------------------------------------------------------------
+// Layer enumeration functions
+// ----------------------------------------------------------------------------
+VKAPI_ATTR VkResult VKAPI_CALL
+OrbitEnumerateInstanceLayerProperties(uint32_t* property_count, VkLayerProperties* properties) {
+  // Vulkan spec dictates that we are only supposed to enumerate ourself
+  if (property_count != nullptr) {
+    *property_count = 1;
+  }
+  if (properties != nullptr) {
+    snprintf(properties->layerName, strlen(kLayerName), kLayerName);
+    snprintf(properties->description, strlen(kLayerDescription), kLayerDescription);
+    properties->implementationVersion = kLayerImplVersion;
+    properties->specVersion = kLayerSpecVersion;
+  }
+
+  return VK_SUCCESS;
+}
+
+// Deprecated by Khronos, but we'll support it in case older applications still
+// use it.
+VKAPI_ATTR VkResult VKAPI_CALL OrbitEnumerateDeviceLayerProperties(
+    VkPhysicalDevice /*physical_device*/, uint32_t* property_count, VkLayerProperties* properties) {
+  // This function is supposed to return the same results as
+  // EnumerateInstanceLayerProperties since device layers were deprecated.
+  return OrbitEnumerateInstanceLayerProperties(property_count, properties);
+}
+
+VKAPI_ATTR VkResult VKAPI_CALL OrbitEnumerateInstanceExtensionProperties(
+    const char* layer_name, uint32_t* property_count, VkExtensionProperties* /*properties*/) {
+  // Inform the client that we have no extension properties if this layer
+  // specifically is being queried.
+  if (layer_name != nullptr && strcmp(layer_name, kLayerName) == 0) {
+    if (property_count != nullptr) {
+      *property_count = 0;
+    }
+    return VK_SUCCESS;
+  }
+
+  // Vulkan spec mandates returning this when this layer isn't being queried.
+  return VK_ERROR_LAYER_NOT_PRESENT;
+}
+
+VKAPI_ATTR VkResult VKAPI_CALL OrbitEnumerateDeviceExtensionProperties(
+    VkPhysicalDevice physical_device, const char* layer_name, uint32_t* property_count,
+    VkExtensionProperties* properties) {
+  // pass through any queries that aren't to us
+  if (layer_name == nullptr || strcmp(layer_name, kLayerName) != 0) {
+    if (physical_device == VK_NULL_HANDLE) {
+      return VK_SUCCESS;
+    }
+
+    return logic_.CallEnumerateDeviceExtensionProperties(physical_device, layer_name,
+                                                         property_count, properties);
+  }
+
+  // don't expose any extensions
```

Comment:
```suggestion
  // Don't expose any extensions.
```

Reply:
And I think this needs more explanation. 

"This layer does not expose any extensions, therefore we need to set property_count to zero."